lib/utils/oauth/src/index.js

```diff
@@ -262,32 +262,40 @@ const cache = (authServer, id, secret, scopes) => {
       done(err);
     };
 
-    // Get OAuth bearer access token
-    const get = () => {
+    const getToken = (endpoint) => {
       debug('Getting OAuth bearer access token from endpoint %s for ' +
-        'client %s and scopes %s', authServer, id, scopes);
-      // Get token endpoint from OAuth server
-      tokenEndpoint(authServer, (err, endpoint) => {
+        'client %s and scopes %s', endpoint, id, scopes);
+
+      newToken(endpoint, method, path, id, secret, scopes, (err, t) => {
         if (err) {
           error(err);
           return;
         }
 
-        // Get token from the OAuth token endpoint
-        newToken(endpoint, id, secret, scopes, (err, t) => {
+        debug('Got OAuth bearer access token');
+        token = t;
+        // Convert token expiration period from seconds to milliseconds
+        // Without conversion, the token will refresh every 2 minutes
+        refresh(Math.max(token.expires_in * 1000 - 120000, 120000));
```

Comment:
According to OAuth 2.0 standard:
`expires_in: The lifetime in seconds of the access token.`
[see here for more](https://tools.ietf.org/html/rfc6749#section-4.2.2)